WolvenKit.Core/Extensions/StringExtensions.cs

```diff
@@ -136,16 +136,35 @@ public static bool IsNullOrEmptyOrEndsWith(this string? target, string value) =>
         public static bool IsEmptyOrEndsWith(this string target, string value) =>
             target == "" || target.EndsWith(value);
 
-#pragma warning disable SYSLIB1045
-        // Generated regex attribute means the class can't be abstract anymore
+
+        private static char[]? s_invalidFileNameChars = null;
+
+        /// <summary>
+        /// Separate method to avoid SYSLIB1045 warning
+        /// </summary>
+        private static char[] GetInvalidFileNameChars()
+        {
+            s_invalidFileNameChars ??= Path.GetInvalidFileNameChars();
+            return s_invalidFileNameChars;
+        }
+        
+        /// <summary>
+        /// Generates redengine friendly file path. 
+        /// </summary>
+        public static string ToFilePath(this string target) => string.Join(Path.DirectorySeparatorChar,
+            target.Split(Path.DirectorySeparatorChar).Select(s => s.ToFileName()));
+
         /// <summary>
         /// Generates redengine friendly file name 
         /// </summary>
         public static string ToFileName(this string target) =>
-            new string(target.Where(c => !Path.GetInvalidFileNameChars().Contains(c)).ToArray()).Trim()
+            new string(target.Where(c => GetInvalidFileNameChars().Contains(c)).ToArray()).Trim()
```

Comment:
why check here for GetInvalidFileNameChars? `new string(target.Where(c => GetInvalidFileNameChars().Contains(c)).ToArray()).Trim()` I don't get it

Reply:
This drops all characters that are not supposed to be in a file name from the string, e.g. backslashes or ascii farts. I've refactored the call to `Path` to get rid of the ReSharper warning.